app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,120 @@
+class GroupsController < ApplicationController
+  before_action :check_access
+  before_action :set_group, only: [:show, :edit, :update, :destroy]
+  after_action :assign_group_patients, only: [:create]
+
+  def index
+    @groups = Group.order(:id).page(params[:page]).per(25)
+  end
+
+  # PATCH/PUT /groups/1
+  def update
+    new_grp_name = params[:group]['name']
+    new_members = params[:users]
+    remove_members = params[:members]
+
+    ActiveRecord::Base.transaction do
+      @group.name = new_grp_name
+      if new_members
+        new_members.each do |m|
+          user = User.where(id: m).first
+          Member.create(user_id: user.id, group_id: @group.id)
+          assign_group_patients
+        end
+      end
+
+      if remove_members
+        remove_members.each do |mem|
+          user = User.where(id: mem).first
+          Member.where(user_id: user.id).where(group_id: @group.id).destroy_all
+          email = user.email
+          submitter = Submitter.find_by_email(email)
+          if submitter != nil
+            submitter.patient.each do |patient|
+              GroupsPatient.where(group_id: @group.id).where(patient_id: patient.id).destroy_all
+            end
+          end
+        end
+      end
+    end
+    respond_to do |format|
+      if @group.save
+        format.html { redirect_to @group, notice: 'Group was Updated successfully' }
+      else
+        format.html { render :new }
+        format.json { render json: @group.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # GET /groups/new
+  def new
+    @groups = Group.new
+    @users = User.all
+  end
+
+  def create
+    grp_name = params[:name]
+    members = params[:users]
+    puts members
+    ActiveRecord::Base.transaction do
+      @group = Group.create(name: grp_name)
+      members.each do |m|
+        user = User.where(id: m).first
+        Member.create(user_id: user.id, group_id: @group.id)
+      end
+    end
+    respond_to do |format|
+      if @group.save
+        format.html { redirect_to @group, notice: 'Group was successfully created' }
+
+      else
+        format.html { render :new }
+        format.json { render json: @group.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def show
+  end
+
+  # GET /patients/1/edit
+  def edit
+    @users = User.all
+    @members = @group.users
+  end
+
+  def destroy
+    @group.destroy
+    respond_to do |format|
+      format.html { redirect_to groups_url, notice: 'Group was successfully destroyed' }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+
+  def set_group
+    @group = Group.find(params[:id])
+  end
+
+  def check_access
+    (current_user.nil?) ? redirect_to(root_path) : (redirect_to(root_path) unless current_user.admin?)
+  end
+
+  def assign_group_patients
```

Comment:
Could you add a column in member table to check if this user wants to share the patients?  I think it will assign all the patients from all members to this group. The use case will be Alexej help the doctor from Leipzig to analyze her patients, but the doctor from Leipzig should not be able to access Alexej's patients. That's why I choose to assign the patients to user_patients before. Sometimes, someone wants to access the case from another person, but he doesn't want to share his patients.